File: ruby/classes_search.rb
require 'json'
require 'open-uri'

REST_URL = "http://stagedata.bioontology.org"
API_KEY = ""

def get_json(url)
  JSON.parse(open(url, "Authorization" => "apikey token=#{API_KEY}").read)
end

# Get list of search terms
path = File.expand_path('../classes_search_terms.txt', __FILE__)
terms_file = open(path, "r")
terms = []
terms_file.each {|line| terms << line}

# Do a search for every term
search_results = []
terms.each do |term|
  search_results << get_json(REST_URL + "/search?q=" + term)["collection"]
end
    
# Print the results
require 'pp'
pp search_results

File: python/python3/classes_search.py
import urllib.request, urllib.error, urllib.parse
import json
import os
from pprint import pprint

REST_URL = "http://data.bioontology.org"
API_KEY = ""

def get_json(url):
    opener = urllib.request.build_opener()
    opener.addheaders = [('Authorization', 'apikey token=' + API_KEY)]
    return json.loads(opener.open(url).read())

# Get list of search terms
path = os.path.join(os.path.dirname(__file__), 'classes_search_terms.txt')
terms_file = open(path, "r")
terms = []
for line in terms_file:
    terms.append(line)

# Do a search for every term
search_results = []
for term in terms:
    search_results.append(get_json(REST_URL + "/search?q=" + term)["collection"])

# Print the results
for result in search_results:
    pprint(result)
